app/src/main/java/ch/epfl/favo/view/tabs/addFavor/FavorRequestView.java

```diff
@@ -21,82 +25,97 @@
 
-    Button confirmFavorBtn = rootView.findViewById(R.id.add_button);
-    confirmFavorBtn.setOnClickListener(this);
+    confirmFavorBtn = rootView.findViewById(R.id.request_button);
+    addPictureBtn = rootView.findViewById(R.id.add_picture_button);
+    mImageView = rootView.findViewById(R.id.imageView);
+
+    confirmFavorBtn.setOnClickListener(
+        new View.OnClickListener() {
+          @Override
+          public void onClick(View v) {
+            requestFavor();
+          }
+        });
+
+    addPictureBtn.setOnClickListener(
+        new View.OnClickListener() {
+          @Override
+          public void onClick(View v) {
+            openFileChooser();
+          }
+        });
 
     mGpsTracker = DependencyFactory.getCurrentGpsTracker(getActivity().getApplicationContext());
 
     return rootView;
   }
 
+  public void openFileChooser() {
+    Intent intent = new Intent();
+    intent.setType("image/*");
+    intent.setAction(Intent.ACTION_GET_CONTENT);
+    startActivityForResult(intent, PICK_IMAGE_REQUEST);
+  }
+
   @Override
-  public void onClick(View view) {
-    // The following inspection warning is suppressed. More cases will be added soon.
-    // noinspection SwitchStatementWithTooFewBranches
-    switch (view.getId()) {
-      case R.id.add_button:
-        showSnackbar(getString(R.string.favor_success_msg));
-        EditText titleElem = Objects.requireNonNull(getView()).findViewById(R.id.title);
-        EditText descElem = Objects.requireNonNull(getView()).findViewById(R.id.title);
-        String title = titleElem.getText().toString();
-        String desc = descElem.getText().toString();
-        Location loc = mGpsTracker.getLocation();
-
-        Favor favor = new Favor(title, desc, null, loc, 0);
-        FavorUtil.getSingleInstance().postFavor(favor);
+  public void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
+    super.onActivityResult(requestCode, resultCode, data);
+
+    if (requestCode == PICK_IMAGE_REQUEST
+        && resultCode == RESULT_OK
+        && data != null
+        && data.getData() != null) {
+      mImageUri = data.getData();
+      mImageView.setImageURI(mImageUri);
+    } else {
+      showSnackbar("Try again!");
     }
   }
 
+  private void requestFavor() {
+    showSnackbar(getString(R.string.favor_request_success_msg));
+    EditText titleElem = Objects.requireNonNull(getView()).findViewById(R.id.title);
+    EditText descElem = Objects.requireNonNull(getView()).findViewById(R.id.details);
+    String title = titleElem.getText().toString();
+    String desc = descElem.getText().toString();
+    Location loc = mGpsTracker.getLocation();
+
+    Favor favor = new Favor(title, desc, null, loc, 0);
+    FavorUtil.getSingleInstance().postFavor(favor);
+  }
+
   // Todo: Implement the following functions to verify user input.
 
   // Todo: Try to put this method in a util package and import it here.
-  private void showSnackbar(String errorMessageRes) {
```

Comment:
We have a common tool of showSnackbar actually. import from common tool but with an additional argument or each class define their own showSnackbar, Which one do you prefer?

Reply:
your option is better. I will update it